### PR TITLE
Dfpl 571 LA submission of standalone "Authority to refuse contact with a child in care" application

### PR DESCRIPTION
**Pull request by chak-shing-lo-justice**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFPL-571

### Change description ###
To introduce the same functionality currently available for LA's via the additional application journey for authority to refuse contact for a child in care

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-definition/CaseEventToComplexTypes/enterGrounds/GroundsForRefuseContactWithChild.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    "EventElementLabel": "<h2>Current arrangement for contact</h2>",
+    "DisplayContext": "READONLY"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "GroundsForRefuseContactWithChild",
+    "CaseEventID": "enterGrounds",
+    "CaseFieldID": "groundsForRefuseContactWithChild",
+    "ListElementCode": "personHasContactAndCurrentArrangement",
+    "EventElementLabel": "*State the full names(s) of each person who has contact with each child and the current arrangements for contact",
```
Comment (so99y) [inline ccd-definition/CaseEventToComplexTypes/enterGrounds/GroundsForRefuseContactWithChild.json:17]: typo? `name(s)`

File: ccd-definition/ComplexTypes/CareSupervision/GroundsForRefuseContactWithChild.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "FieldType": "Label",
+    "ElementLabel": "<h2>Current arrangement for contact</h2>",
+    "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "GroundsForRefuseContactWithChild",
+    "ListElementCode": "personHasContactAndCurrentArrangement",
+    "FieldType": "TextArea",
+    "ElementLabel": "State the full names(s) of each person who has contact with each child and the current arrangements for contact",
```
Comment (so99y) [inline ccd-definition/ComplexTypes/CareSupervision/GroundsForRefuseContactWithChild.json:15]: again, `full name(s)`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/GroundsForRefuseContactWithChild.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import lombok.Builder;
+import lombok.Data;
+
+import javax.validation.constraints.NotBlank;
+
+@Data
+@Builder
+@AllArgsConstructor(onConstructor_ = {@JsonCreator})
+public class GroundsForRefuseContactWithChild {
+    @NotBlank(message = "Please state the full names(s) of each person who has contact with each child "
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/GroundsForRefuseContactWithChild.java:14]: name(s)

File: ccd-definition/CaseEventToComplexTypes/enterGrounds/GroundsForRefuseContactWithChild.json (hunk 1 above)
Comment (chak-shing-lo-justice) [reply ccd-definition/CaseEventToComplexTypes/enterGrounds/GroundsForRefuseContactWithChild.json:17]: Nope. Copied it from the pdf of C14

File: ccd-definition/ComplexTypes/CareSupervision/GroundsForRefuseContactWithChild.json (hunk 2 above)
Comment (chak-shing-lo-justice) [reply ccd-definition/ComplexTypes/CareSupervision/GroundsForRefuseContactWithChild.json:15]: Nope. Copied it from the pdf of C14

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/GroundsForRefuseContactWithChild.java (hunk 3 above)
Comment (chak-shing-lo-justice) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/GroundsForRefuseContactWithChild.java:14]: Nope

**Review by so99y — APPROVED** [review]

**Review by AlistairEdwinOsborne — DISMISSED** [review]

**Pull request merged**